backend/scripts/verify_spotify_parquet.py

```
@@ -169,15 +169,15 @@ def verify_r2_data():
-                logger.error(f"❌ DuckDB IO Error (artists): {e}")
+                logger.error("❌ DuckDB IO Error (artists): %s", e)
 
     except duckdb.IOException as e:
         if "No files found" in str(e):
             logger.warning("⚠️ No Parquet files found in the specified path.")
         else:
-            logger.error(f"❌ DuckDB IO Error: {e}")
+            logger.error("❌ DuckDB IO Error: %s", e)
     except Exception as e:
-        logger.error(f"❌ Unexpected Error: {e}")
+        logger.error("❌ Unexpected Error: %s", e)
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)